### PR TITLE
Serve data about available block types in API

### PR DESCRIPTION
**Pull request by ml-evs**
Supersedes #455 which had gone stale. Aiming to push this one just a little further to replicate our current behavior before adding e.g., block docs in the UI.

For now, this simply replicates the current behavior, but begins exposing some block info in the API directly. I also cleaned up some dead routes, test pages and did a fixup of some of the block docstrings.

This info can then be used by e.g., @BenjaminCharmes to begin populating block-level documentation in the app, in preparation for a future broader blocks rework in #616.

### CODE REVIEW

**Review by BenjaminCharmes — APPROVED** [review]
Great!

File: webapp/src/resources.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Resources for the application
-import DataBlockBase from "@/components/datablocks/DataBlockBase";
+import GenericBlock from "@/components/datablocks/GenericBlock";
```
Comment (ml-evs) [inline webapp/src/resources.js:2]:
```suggestion
import DataBlockBase from "@/components/datablocks/DataBlockBase";
```

File: webapp/src/resources.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const UPPY_MAX_NUMBER_OF_FILES =
 export const debounceTime = 250; // time after user stops typing before request is sent
 
 export const blockTypes = {
-  comment: { description: "Comment", component: DataBlockBase },
+  comment: { description: "Comment", component: GenericBlock },
```
Comment (ml-evs) [inline webapp/src/resources.js:48]:
```suggestion
  comment: { description: "Comment", component: DataBlockBase },
```

**Review by BenjaminCharmes — APPROVED** [review]
Looks great !

**Pull request merged**